src/fabric.c

```diff
@@ -1499,6 +1502,11 @@ int DEFAULT_SYMVER_PRE(fi_fabric)(struct fi_fabric_attr *attr,
 			attr->name);
```

Comment:
I would replace this print with a new function that includes the additional information.  E.g. "Opened fabric: blah from provider: blahblah (ver: 1.15) path: blah/blah".  That would isolate the ifdef to the new function.  I included the provider version as that indirectly indicates which provider was picked up.  But I'd also include the path, to make it easier on the user and avoid needing to cross reference this print with an earlier print.  For providers that were built in, I would indicate that in the print, so that it's conveyed directly, rather than the reader needing to know that the absence of a print means build-in,

Reply:
I don't want to combine the two prints and put it under #ifdef HAVE_LIBDL because you will lose the print without DL.  I've added version number as it may be useful.  If the provider is built-in, path to libfabric.so will get printed.

Reply:
Having ifdef's within a function make the maintenance of the function more difficult.  That's why all or nearly all ifdefs we have surround entire functions.  I was suggesting something like:
```
#ifdef HAVE_LIBDL
void ofi_print_fabric_prov(...)
{
    FI_INFO(...)
}
#else
void ofi_print_fabric_prov(...)
{
    FI_INFO(..."internal"...)
}
#endif

Then just call ofi_print_fabric_prov().
```

Reply:
You are bringing back nightmares with irdma_kcompat.h, please refrain from discussing maintainability with #ifdef.  :-)  With that, this one is more inline with the one in ofi_cleanup_prov than the big wrapper around ofi_reg_dl_prov, ofi_ini_dir, etc where those were not called and only applicable with libdl.  To create a function out of a print that's only needed with libdl is overkill.

Reply:
Just to clarify, `HAVE_LIBDL` is almost always true unless the use of DL provider is explicitly disabled (`--with-dlopen=no`). Both built-in providers and DL providers are reported under this code path. The latest patch revision looks fine to me.